pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -481,8 +492,10 @@ public CompletableFuture<Void> unloadSubscription(@Nonnull String subName) {
 
     private PersistentSubscription createPersistentSubscription(String subscriptionName, ManagedCursor cursor,
             boolean replicated, Map<String, String> subscriptionProperties) {
-        Objects.requireNonNull(compactedTopic);
-        if (isCompactionSubscription(subscriptionName)) {
+        Objects.requireNonNull(topicCompactionService);
+        if (isCompactionSubscription(subscriptionName)
+                && topicCompactionService instanceof PulsarTopicCompactionService pulsarTopicCompactionService) {
+            CompactedTopicImpl compactedTopic = pulsarTopicCompactionService.getCompactedTopic();
```

Comment:
What about the compaction service plugin? Will it use the same compaction cursor for maintaining the compaction status?

Reply:
Yes, the compaction cursor (`__compaction`) still be used for maintaining the compaction status.

Reply:
Could you point out how the broker creates the __compaction subscription if the topic compaction service is not `PulsarTopicCompactionService`? Or it will be handled by the user?

Reply:
Other TopicCompactionService also creates the `__compaction` subscription, see: line-496.
In addition, `PulsarCompactorSubscription` only overrides the `aknowledgeMessage` method to early delete historical data, but it is not required the other `TopicCompactionService` can handle it through other way.